### PR TITLE
fix: add suggestion how to fix importing CJS module

### PR DESCRIPTION
**Pull request by bartlomieju**
```
$ cat exports_error.js
Object.defineProperty(exports, "__esModule", { value: true });

$ deno exports_error.js
error: Uncaught (in promise) ReferenceError: exports is not defined
Object.defineProperty(exports, "__esModule", { value: true });
                      ^
    at file:///exports_error.js:1:23

    hint: Deno doesn't support CommonJS modules without `.cjs` extension.
    hint: Rewrite this module to ESM or change the file extension to `.cjs`.
```

### CODE REVIEW

File: cli/main.rs [hunk 1]
```diff
@@ -260,7 +260,16 @@ fn unwrap_or_exit<T>(result: Result<T, AnyError>) -> T {
       let mut error_code = 1;
 
       if let Some(e) = error.downcast_ref::<JsError>() {
+        // eprintln!("is js error! {:?}", e);
         error_string = format_js_error(e);
+
+        if e.name == Some("ReferenceError".to_string()) {
+          if let Some(msg) = &e.message {
+            if msg.contains("module is not defined") {
+              error_string = format!("{}\n\n    {} Deno doesn't support CommonJS modules, change problematic module into an ES module.", error_string, colors::cyan("hint:"))
```
Comment (ry) [inline cli/main.rs:269]: Maybe something like 

> Deno doesn't support CommonJS in user code, that is, outside of npm modules. Consider converting the above file to ES module syntax.

Not so sure about hard coding colors like this. Maybe colors::bold instead?

**Comment by littledivy** [comment]
Really like the idea of providing suggestions.

Here's a little more real-worldy next config: https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/main/next.config.js

Ideally Deno could provide suggestions for require migration:

```
| const { withContentlayer } = require('next-contentlayer')
                               ^^^^^^^
Deno doesn't support CommonJS in user code, that is, outside of npm modules.

hint: use `import { withContentLayer } from "next-contentlayer"`;

| const withBundleAnalyzer = require('@next/bundle-analyzer')({
                             ^^^^^^^

Deno doesn't support CommonJS in user code, that is, outside of npm modules.

hint: use:
          import $ from "@next/bundle-analyzer";
          const withBundleAnalyzer = $({
            enabled: process.env.ANALYZE === 'true',
          });
```

**Comment by scarf005** [comment]
> Ideally Deno could provide suggestions for migration:

being able to have a CLI option to autofix given hint would really be helpful.

**Comment by bartlomieju** [comment]
Update the error message and land, but first do https://github.com/denoland/deno/issues/25384

File: tests/specs/run/import_common_js/exports_error.out [hunk 2]
```diff
@@ -0,0 +1,7 @@
+error: Uncaught (in promise) ReferenceError: exports is not defined
+Object.defineProperty(exports, "__esModule", { value: true });
+                      ^
+    at [WILDCARD]exports_error.js:1:23
+
+    hint: Deno doesn't support CommonJS modules without `.cjs` extension.
+    hint: Rewrite this module to ESM or change the file extension to `.cjs`.
```
Comment (dsherret) [inline tests/specs/run/import_common_js/exports_error.out:7]: Might be nice as:

```diff
-   hint: Deno doesn't support CommonJS modules without `.cjs` extension.
+   info: Deno doesn't support CommonJS modules without `.cjs` extension.
    hint: Rewrite this module to ESM or change the file extension to `.cjs`.
```

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/main.rs [hunk 3]
```diff
@@ -336,12 +336,32 @@ fn exit_with_message(message: &str, code: i32) -> ! {
...
 }
 
+fn get_hints_for_commonjs_error(e: &JsError) -> Vec<String> {
+  if e.name.as_deref() == Some("ReferenceError") {
+    if let Some(msg) = &e.message {
+      if msg.contains("module is not defined")
+        || msg.contains("exports is not defined")
+      {
+        return vec![
+          "Deno doesn't support CommonJS modules without `.cjs` extension."
```
Comment (dsherret) [inline cli/main.rs:346]:
```suggestion
          "Deno does not support CommonJS modules without `.cjs` extension."
```

**Pull request merged**